testworld/signature_test.go

```
@@ -0,0 +1,146 @@
+// +build testworld
+
+package testworld
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/centrifuge/centrifuge-protobufs/gen/go/coredocument"
+	"github.com/centrifuge/go-centrifuge/config/configstore"
+	"github.com/centrifuge/go-centrifuge/contextutil"
+	"github.com/centrifuge/go-centrifuge/crypto"
+	"github.com/centrifuge/go-centrifuge/documents"
+	"github.com/centrifuge/go-centrifuge/documents/purchaseorder"
+	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/centrifuge/go-centrifuge/testingutils/config"
+	"github.com/centrifuge/go-centrifuge/testingutils/documents"
+	"github.com/centrifuge/go-centrifuge/utils"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHost_ValidSignature(t *testing.T) {
+	t.Parallel()
+
+	// Hosts
+	bob := doctorFord.getHostTestSuite(t, "Bob")
+	eve := doctorFord.getHostTestSuite(t, "Eve")
+
+	ctxh := testingconfig.CreateAccountContext(t, eve.host.config)
+
+	collaborators := [][]byte{bob.id[:]}
+	dm := createCDWithEmbeddedPO(t, collaborators, eve.id, ctxh)
+	assert.Equal(t, 1, len(dm.Signatures()))
+
+	signatures, _, _ := eve.host.p2pClient.GetSignaturesForDocument(ctxh, dm)
```

don't ignore the errors. instead, check for `nil` errors